src/osd/ReplicatedPG.cc

```diff
@@ -2240,14 +2240,15 @@ bool ReplicatedPG::maybe_promote(ObjectContextRef obc,
   dout(20) << __func__ << " missing_oid " << missing_oid
```

Comment:
No.. promote_object() does the scrub check and may put promote_op on waiting_for_active.  That needs to move into queue_async_promote.

...but then this interacts poorly with _rollback_to.  You need to do an unconditoinal promote if promote_obc != nullptr.